analysis/druid/src/core/ActiveDruidForm.ts

```diff
@@ -0,0 +1,74 @@
+
+    constructor(options: Options) {
+        super(options);
+        this.active = true
+        
+        //Balance
+        this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.MOONKIN_FORM_AFFINITY, SPELLS.MOONKIN_FORM]), this.moonkinActivated)
+        this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell([SPELLS.MOONKIN_FORM_AFFINITY, SPELLS.MOONKIN_FORM]), this.noformActivated)
+        this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.INCARNATION_CHOSEN_OF_ELUNE_TALENT]), this.incarnMoonkinActivated)
+        this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell([SPELLS.INCARNATION_CHOSEN_OF_ELUNE_TALENT]), this.noformActivated)
+
+        //Feral
+        this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.CAT_FORM]), this.catActivated)
+        this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell([SPELLS.CAT_FORM]), this.noformActivated)
+        this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.INCARNATION_KING_OF_THE_JUNGLE_TALENT]), this.incarnCatActivated)
+        this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell([SPELLS.INCARNATION_KING_OF_THE_JUNGLE_TALENT]), this.noformActivated)
+
+        //Guardian
+        this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.BEAR_FORM]), this.bearActivated)
+        this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell([SPELLS.BEAR_FORM]), this.noformActivated)
+        this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.INCARNATION_GUARDIAN_OF_URSOC_TALENT]), this.incarnBearActivated)
+        this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell([SPELLS.INCARNATION_GUARDIAN_OF_URSOC_TALENT]), this.noformActivated)
+    
+        //Resto
+        this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.INCARNATION_TREE_OF_LIFE_TALENT]), this.incarnTOLActivated)
+        this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell([SPELLS.INCARNATION_TREE_OF_LIFE_TALENT]), this.noformActivated)
+
+    }
+
+    get form() {
+        return this.currentForm;
+    }
+
+    incarnTOLActivated() {
+        this.currentForm = "Tree of Life"
+    }
+
+    incarnBearActivated() {
+        this.currentForm = "Incarnation: Bear"
+    }
+
+    incarnCatActivated() {
+        this.currentForm = "Incarnation: Cat"
+    }
+
+    incarnMoonkinActivated() {
+        this.currentForm = "Incarnation: Moonkin"
+    }
+
+    moonkinActivated() {
+        this.currentForm = "Moonkin"
+    }
+
+    bearActivated() {
+        this.currentForm = "Bear"
+    }
+
+    catActivated() {
+        this.currentForm = "Cat"
+    }
+
+    noformActivated() {
+        this.currentForm = "No Form"
+    }
+}
+
+export type DruidForm = "Cat" | "Moonkin" | "Bear" | "No Form" | "Tree of Life" | "Incarnation: Cat" | "Incarnation: Bear" | "Incarnation: Moonkin"
```

Comment:
There is inconsistency throughout this file on trailing semicolons. I thought there was an eslint rule to always include them, but can't find it now. Regardless, I'd try to keep it consistent whichever way you go.

Reply:
I've decided to go with the semicolons, should be fixed now.